diff -ruN /home/mdamico/devel/polkadot-sdk/polkadot/node/core/pvf/common/Cargo.toml common/Cargo.toml
--- /home/mdamico/devel/polkadot-sdk/polkadot/node/core/pvf/common/Cargo.toml	2025-12-09 17:18:59.915571108 +0100
+++ common/Cargo.toml	2025-12-10 11:24:58.483901820 +0100
@@ -4,41 +4,39 @@
 version = "17.0.1"
 authors.workspace = true
 edition.workspace = true
-license.workspace = true
+license = "GPL-3.0-only"
 
 [lints]
 workspace = true
 
 [dependencies]
-cpu-time = { workspace = true }
-futures = { workspace = true }
-gum.workspace = true
-gum.default-features = true
-libc = { workspace = true }
-nix = { features = ["resource", "sched"], workspace = true }
-thiserror = { workspace = true }
+native = { workspace = true }
 
-codec = { features = ["derive"], workspace = true }
+cpu-time = "1.0.0"
+futures = { workspace = true }
+libc = "0.2.172"
+nix = { version = "0.28.0", features = ["resource", "sched"] }
+thiserror = "1.0.64"
+
+codec = {version = "3.6.12", package = "parity-scale-codec", default-features = false, features = [
+ "derive",
+] }
+gum = { version = "17.0.0", package = "tracing-gum", default-features = true }
 polkadot-parachain-primitives.workspace = true
 polkadot-parachain-primitives.default-features = true
 polkadot-primitives.workspace = true
 polkadot-primitives.default-features = true
 sc-executor.workspace = true
 sc-executor.default-features = true
-sc-executor-common.workspace = true
-sc-executor-common.default-features = true
-sc-executor-wasmtime.workspace = true
-sc-executor-wasmtime.default-features = true
+sc-executor-common = { version = "0.36.0", default-features = true }
+sc-executor-wasmtime = { version = "0.36.0", default-features = true }
 sp-core.workspace = true
 sp-core.default-features = true
-sp-crypto-hashing.workspace = true
-sp-crypto-hashing.default-features = true
-sp-externalities.workspace = true
-sp-externalities.default-features = true
+sp-crypto-hashing = { version = "0.1.0", default-features = true }
+sp-externalities = { version = "0.30.0", default-features = true }
 sp-io.workspace = true
 sp-io.default-features = true
-sp-tracing.workspace = true
-sp-tracing.default-features = true
+sp-tracing = { version = "17.0.1", default-features = true }
 
 [target.'cfg(target_os = "linux")'.dependencies]
 landlock = "0.3.0"
@@ -47,7 +45,7 @@
 seccompiler = "0.4.0"
 
 [dev-dependencies]
-assert_matches = { workspace = true }
+assert_matches = "1.5.0"
 
 [target.'cfg(target_os = "linux")'.dev-dependencies]
 tempfile = { workspace = true }
diff -ruN /home/mdamico/devel/polkadot-sdk/polkadot/node/core/pvf/common/rustfmt.toml common/rustfmt.toml
--- /home/mdamico/devel/polkadot-sdk/polkadot/node/core/pvf/common/rustfmt.toml	1970-01-01 01:00:00.000000000 +0100
+++ common/rustfmt.toml	2024-12-03 17:04:41.764197000 +0100
@@ -0,0 +1 @@
+disable_all_formatting = true
diff -ruN /home/mdamico/devel/polkadot-sdk/polkadot/node/core/pvf/common/src/executor_interface.rs common/src/executor_interface.rs
--- /home/mdamico/devel/polkadot-sdk/polkadot/node/core/pvf/common/src/executor_interface.rs	2024-12-16 13:55:52.322595354 +0100
+++ common/src/executor_interface.rs	2025-12-10 11:15:03.875783496 +0100
@@ -208,6 +208,7 @@
 	sp_io::allocator::HostFunctions,
 	sp_io::logging::HostFunctions,
 	sp_io::trie::HostFunctions,
+	native::HLNativeHostFunctions,
 );
 
 /// The validation externalities that will panic on any storage related access. (PVFs should not
