diff -ruN /home/daniele/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/polkadot-primitives-17.1.0/src/runtime_api.rs patches/polkadot/primitives/src/runtime_api.rs
--- /home/daniele/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/polkadot-primitives-17.1.0/src/runtime_api.rs	2006-07-24 03:21:28.000000000 +0200
+++ patches/polkadot/primitives/src/runtime_api.rs	2026-02-25 15:11:31.158137972 +0100
@@ -116,7 +116,8 @@
 use crate::{
 	slashing,
 	vstaging::{
-		self, CandidateEvent, CommittedCandidateReceiptV2 as CommittedCandidateReceipt, CoreState,
+      		self, async_backing::Constraints, CandidateEvent,
+		CommittedCandidateReceiptV2 as CommittedCandidateReceipt, CoreState,
 		ScrapedOnChainVotes,
 	},
 	ApprovalVotingParams, AsyncBackingParams, BlockNumber, CandidateCommitments, CandidateHash,
@@ -302,5 +303,14 @@
 		/// Retrieve the maximum uncompressed code size.
 		#[api_version(12)]
 		fn validation_code_bomb_limit() -> u32;
+
+		/// Returns the constraints on the actions that can be taken by a new parachain
+		/// block.
+		#[api_version(13)]
+		fn backing_constraints(para_id: ppp::Id) -> Option<Constraints>;
+
+		/// Retrieve the scheduling lookahead
+		#[api_version(13)]
+		fn scheduling_lookahead() -> u32;
 	}
 }
diff -ruN /home/daniele/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/polkadot-primitives-17.1.0/src/vstaging/async_backing.rs patches/polkadot/primitives/src/vstaging/async_backing.rs
--- /home/daniele/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/polkadot-primitives-17.1.0/src/vstaging/async_backing.rs	2006-07-24 03:21:28.000000000 +0200
+++ patches/polkadot/primitives/src/vstaging/async_backing.rs	2026-02-24 12:16:13.096520380 +0100
@@ -50,12 +50,51 @@
 	}
 }
 
+/// Constraints on the actions that can be taken by a new parachain
+/// block. These limitations are implicitly associated with some particular
+/// parachain, which should be apparent from usage.
+#[derive(RuntimeDebug, Clone, PartialEq, Encode, Decode, TypeInfo)]
+pub struct Constraints<N = BlockNumber> {
+	/// The minimum relay-parent number accepted under these constraints.
+	pub min_relay_parent_number: N,
+	/// The maximum Proof-of-Validity size allowed, in bytes.
+	pub max_pov_size: u32,
+	/// The maximum new validation code size allowed, in bytes.
+	pub max_code_size: u32,
+	/// The maximum head-data size, in bytes.
+	pub max_head_data_size: u32,
+	/// The amount of UMP messages remaining.
+	pub ump_remaining: u32,
+	/// The amount of UMP bytes remaining.
+	pub ump_remaining_bytes: u32,
+	/// The maximum number of UMP messages allowed per candidate.
+	pub max_ump_num_per_candidate: u32,
+	/// Remaining DMP queue. Only includes sent-at block numbers.
+	pub dmp_remaining_messages: Vec<N>,
+	/// The limitations of all registered inbound HRMP channels.
+	pub hrmp_inbound: InboundHrmpLimitations<N>,
+	/// The limitations of all registered outbound HRMP channels.
+	pub hrmp_channels_out: Vec<(Id, OutboundHrmpChannelLimitations)>,
+	/// The maximum number of HRMP messages allowed per candidate.
+	pub max_hrmp_num_per_candidate: u32,
+	/// The required parent head-data of the parachain.
+	pub required_parent: HeadData,
+	/// The expected validation-code-hash of this parachain.
+	pub validation_code_hash: ValidationCodeHash,
+	/// The code upgrade restriction signal as-of this parachain.
+	pub upgrade_restriction: Option<UpgradeRestriction>,
+	/// The future validation code hash, if any, and at what relay-parent
+	/// number the upgrade would be minimally applied.
+	pub future_validation_code: Option<(N, ValidationCodeHash)>,
+}
+
+
 /// The per-parachain state of the backing system, including
 /// state-machine constraints and candidates pending availability.
 #[derive(RuntimeDebug, Clone, PartialEq, Encode, Decode, TypeInfo)]
 pub struct BackingState<H = Hash, N = BlockNumber> {
 	/// The state-machine constraints of the parachain.
-	pub constraints: Constraints<N>,
+	pub constraints: crate::async_backing::Constraints<N>,
 	/// The candidates pending availability. These should be ordered, i.e. they should form
 	/// a sub-chain, where the first candidate builds on top of the required parent of the
 	/// constraints and each subsequent builds on top of the previous head-data.
diff -ruN /home/daniele/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/polkadot-primitives-17.1.0/src/vstaging/mod.rs patches/polkadot/primitives/src/vstaging/mod.rs
--- /home/daniele/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/polkadot-primitives-17.1.0/src/vstaging/mod.rs	2006-07-24 03:21:28.000000000 +0200
+++ patches/polkadot/primitives/src/vstaging/mod.rs	2026-02-24 12:16:35.802010608 +0100
@@ -19,10 +19,11 @@
 
 // Put any primitives used by staging APIs functions here
 use super::{
-	async_backing::Constraints, BlakeTwo256, BlockNumber, CandidateCommitments,
-	CandidateDescriptor, CandidateHash, CollatorId, CollatorSignature, CoreIndex, GroupIndex, Hash,
-	HashT, HeadData, Header, Id, Id as ParaId, MultiDisputeStatementSet, ScheduledCore,
-	UncheckedSignedAvailabilityBitfields, ValidationCodeHash,
+    async_backing::{InboundHrmpLimitations, OutboundHrmpChannelLimitations},
+	BlakeTwo256, BlockNumber, CandidateCommitments, CandidateDescriptor, CandidateHash, CollatorId,
+	CollatorSignature, CoreIndex, GroupIndex, Hash, HashT, HeadData, Header, Id, Id as ParaId,
+	MultiDisputeStatementSet, ScheduledCore, UncheckedSignedAvailabilityBitfields,
+	UpgradeRestriction, ValidationCodeHash,
 };
 use alloc::{
 	collections::{BTreeMap, BTreeSet, VecDeque},
